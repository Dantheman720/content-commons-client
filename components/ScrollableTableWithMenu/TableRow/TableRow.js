import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';
import MyProjectPrimaryCol from 'components/admin/Dashboard/MyProjects/MyProjectPrimaryCol/MyProjectPrimaryCol';
import TableMobileDataToggleIcon from 'components/ScrollableTableWithMenu/TableMobileDataToggleIcon/TableMobileDataToggleIcon';

const TableRow = props => {
  const {
    d, selectedItems, tableHeaders, toggleItemSelection
  } = props;

  return (
    <Table.Row
      className={ d.isNew ? 'myProjects_newItem' : '' }
    >
      { tableHeaders.map( ( header, i ) => (
        <Table.Cell
          data-header={ header.label }
          key={ `${d.id}_${header.name}` }
          className="items_table_item"
        >
          { i === 0
            ? (
              // Table must include .primary_col div for fixed column
              <Fragment>
                <div className="primary_col">
                  <MyProjectPrimaryCol
                    d={ d }
                    header={ header }
                    selectedItems={ selectedItems }
                    toggleItemSelection={ toggleItemSelection }
                  />
                </div>
                <TableMobileDataToggleIcon />
              </Fragment>
            )
            : (
              <Fragment>
                <span>
                  <div className="items_table_mobileHeader">{ header.label }</div>
                  { d[header.name] }
                </span>
                <br />
                { header.label === 'MODIFIED'
                  ? <span>{ d.status }</span>
                  : null }
              </Fragment>
            ) }
        </Table.Cell>
      ) ) }
    </Table.Row>
  );
};

TableRow.propTypes = {
  d: PropTypes.object,
  selectedItems: PropTypes.object,
  tableHeaders: PropTypes.array,
  toggleItemSelection: PropTypes.func
};

export default TableRow;
