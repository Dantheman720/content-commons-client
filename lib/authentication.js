
const allowedRolesForRestrictedPages = ['EDITOR', 'TEAM_ADMIN', 'ADMIN'];

export const isRestrictedPage = path => path.indexOf( '/admin/' ) !== -1;

/**
   *  Checks for valid permissions
   * @param {array} permissions user permissions
   * @param {array} allowed allowed roles to access restricted pages
   */
export const hasPermissionsToAccessPage = permissions => {
  if ( !Array.isArray( permissions ) ) {
    return false;
  }
  return permissions.some( permission => allowedRolesForRestrictedPages.includes( permission ) );
};
