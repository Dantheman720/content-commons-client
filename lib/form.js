const getErrorsFromValidationError = validationError => {
  const FIRST_ERROR = 0;
  return validationError.inner.reduce( ( errors, error ) => ( {
    ...errors,
    [error.path]: error.errors[FIRST_ERROR],
  } ), {} );
};

export const validate = getValidationSchema => values => {
  const validationSchema = getValidationSchema( values );
  try {
    validationSchema.validateSync( values, { abortEarly: false } );
    return {};
  } catch ( error ) {
    return getErrorsFromValidationError( error );
  }
};

export const optionFormatter = options => options.map( option => ( {
  key: option.name, text: option.name, value: option.name
} ) );


// Need to use this callback instead of the standard formik handleChange
// to deal with groups, i.e. selects, dropdowns, radios, as default formik handleChange
// doesn't set correct value
export const formikHandleOnChange = ( name, value, setFieldValue ) => setFieldValue( name, value );


// Need to use this callback instead of the standard formik handleChange
// to deal with with groups, i.e. checkboxes, as default formik handleChange
// doesn't set correct value
export const formikHandleCheckboxOnChange = ( data, setFieldValue, values ) => {
  const { name, value, checked } = data;
  const valueArray = [...values.contentType];

  if ( checked ) {
    valueArray.push( value );
  } else {
    valueArray.splice( valueArray.indexOf( value ), 1 );
  }
  setFieldValue( name, valueArray );
};
