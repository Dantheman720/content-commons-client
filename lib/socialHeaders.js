/**
 * Check if content exists for a given property
 * If so, it is pushed to a properties array
 *
 * @param {object} array Target array into which values are added
 * @param {string} property Property name
 * @param {string} content Property content
 */
const checkAndPush = ( array, property, content ) => {
  if ( content ) {
    const newMeta = {
      property,
      content
    };

    array.push( newMeta );
  }
};

/**
 * Converts the content item object into a metavalues array using the checkAndPush method
 *
 * @param {object} item Object of metadata for a given content item
 * @param {string} url The URL of the page to be rendered
 * @returns {array} Array of { property: content } key pairs
 */
export const populateMetaArray = ( item, url ) => {
  const {
    // description,
    thumbnail,
    selectedLanguageUnit,
    title,
    type
  } = item;

  /* TEMP vars for Document use case */
  const trimContent = item.content && `${item.content.rawText.substring( 0, 152 )}...`;
  const description = item.description || trimContent;

  const videoSrc = ( selectedLanguageUnit && selectedLanguageUnit.source && selectedLanguageUnit.source[0] )
    ? selectedLanguageUnit.source[0]
    : {};
  const streamSrc = ( videoSrc.stream && videoSrc.stream.link ) ? videoSrc.stream.link : '';

  // Initialize metavalue array with universal properties
  const metaArr = [
    {
      property: 'og:type',
      content: ( type && type === 'video' ) ? 'video.other' : 'article'
    },
    {
      property: 'og:url',
      content: url
    },
    {
      property: 'twitter:card',
      content: ( type && type === 'video' ) ? 'player' : 'summary'
    }
  ];

  // Temporary array of optional meta values
  const propsArray = [
    {
      property: 'og:description',
      content: description
    },
    {
      property: 'og:image',
      content: thumbnail
    },
    {
      property: 'og:title',
      content: title
    },
    {
      property: 'og:video:url',
      content: streamSrc
    },
    {
      property: 'twitter:player',
      content: streamSrc
    },
    {
      property: 'twitter:player:height',
      content: ( type && type === 'video' ) ? '196px' : ''
    },
    {
      property: 'twitter:player:width',
      content: ( type && type === 'video' ) ? '350px' : ''
    }
  ];

  // Checks for desired meta tags against items
  propsArray.forEach( propPair => {
    checkAndPush( metaArr, propPair.property, propPair.content );
  } );

  return metaArr;
};
