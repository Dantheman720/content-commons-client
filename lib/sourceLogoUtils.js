import logoDOS from 'static/images/dos_seal.svg';

const dosOwners = [
  'GPA Video',
  'GPA Media Strategy',
  'U.S. Missions',
];

export const displayDOSLogo = teamOrOwnerName => {
  if ( dosOwners.includes( teamOrOwnerName ) ) {
    return logoDOS;
  }
  return null;
};
