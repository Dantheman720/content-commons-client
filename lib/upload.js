import axios from 'axios';
import getConfig from 'next/config';
import mime from 'mime-types';

const { publicRuntimeConfig } = getConfig();

const VIMEO_HEADERS = {
  Authorization: `bearer ${publicRuntimeConfig.REACT_APP_VIMEO_TOKEN}`,
  'Content-Type': 'application/json',
  Accept: 'application/vnd.vimeo.*+json;version=3.4'
};

export const uploadToVimeo = async file => {
  const res = await axios.post( 'https://api.vimeo.com/me/videos', {
    upload: {
      approach: 'pull',
      size: file.input.size,
      link: `https://s3.amazonaws.com/${publicRuntimeConfig.REACT_APP_AWS_S3_PUBLISHER_UPLOAD_BUCKET}/${file.s3Path}`
    }
  }, {
    headers: VIMEO_HEADERS
  } ).catch( err => console.dir( err ) );

  return res.data.link;
};

export const uploadToS3 = async ( projectId, file, getSignedS3Url, callback ) => {
  const { input } = file;
  const contentType = input.type ? input.type : mime.lookup( input.name );

  // 1. get signed url from aws
  const res = await getSignedS3Url( {
    variables: {
      contentType,
      filename: input.name,
      projectId
    }
  } ).catch( err => console.dir( err ) );

  const signedUrl = res.data.getSignedS3Url.url;

  // 2. use signed url to upload to s3 directly
  await axios.put( signedUrl, input, {
    headers: {
      'Content-Type': contentType
    },
    onUploadProgress: async progressEvent => {
      callback( progressEvent, file );
    }
  } ).catch( err => console.dir( err ) );

  const result = {
    path: res.data.getSignedS3Url.key,
    contentType
  };

  return result;
};


/**
   * Return file metadata, ie. duration, bitrate, width, height, etc
   * @param {String} path url to publically available file
 */
export const getFileMetadata = async ( getFileInfo, path ) => {
  const res = await getFileInfo( {
    variables: {
      path
    }
  } ).catch( err => console.dir( err ) );

  if ( res && res.data ) {
    const {
      duration, bitrate, width, height
    } = res.data.getFileInfo;

    return {
      duration,
      bitrate,
      width,
      height
    };
  }
  return null;
};
