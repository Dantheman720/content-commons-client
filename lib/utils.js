import config from 'config';

export const millisToSeconds = millis => {
  if ( typeof millis !== 'number' ) {
    throw new Error( '_millisToSeconds(): Provided parameter is not a number' );
  }
  return ( ( millis % 60000 ) / 1000 ).toFixed( 2 );
};

export const numberWithCommas = number => {
  if ( typeof number !== 'number' ) {
    throw new Error( 'Error: Parameter provided is not a number' );
  }
  return number.toString().replace( /\B(?=(\d{3})+(?!\d))/g, ',' );
};

/**
 * Escape the id if the first letter is a reserved character
 * @param {string } id ES _id
 */
export const escape = id => {
  const re = /[+\-=!(){}[\]"~*?:\\^/]/;
  return re.test( id.charAt( 0 ) ) ? `\\${id}` : id;
};

export const escapeRegExp = string => string.replace( /[.*+-=&!?^~${}()|[\]\\]/g, '\\$&' );

/**
 * If there are an odd number of quotes in the query string is will cause
 * an error. So in that case a string is returned with some quotes escaped
 * to prevent error.
 *
 * @param str
 * @returns string
 */
export const maybeFixQuotes = str => {
  const quoteCount = str.replace( /[^"]/g, '' ).length;

  // If only 1 quote then escape it
  if ( quoteCount === 1 ) return str.replace( '"', '\\"' );
  // If an odd number of quotes then escape all the ones in the middle
  if ( quoteCount % 2 === 1 ) {
    const parts = str.split( '"' );
    return `${parts[0]}"${parts.slice( 1, -1 ).join( '\\"' )}"${parts[parts.length - 1]}`;
  }
  return str;
};

export const capitalizeFirst = str => str.substr( 0, 1 ).toUpperCase() + str.substr( 1 );
export const titleCase = str => str
  .toLowerCase()
  .split( ' ' )
  .map( word => word.charAt( 0 ).toUpperCase() + word.slice( 1 ) )
  .join( ' ' );

/**
 *
 * @param {*} timeSinceLastLoad number in milliseconds time since last saved, generally used to check redux store
 */
export const isDataStale = timeSinceLastLoad => ( Date.now() - timeSinceLastLoad ) > config.TIME_TO_STALE;

export const createHashMap = ( array, key ) => (
  array.reduce( ( acc, obj ) => ( {
    ...acc,
    [obj[key]]: obj
  } ), {} )
);

/**
 * Fetch id fron url. A Youtube link can either use the
 * short form or long form so check both
 * @param {string} url youtube share url
 * @return youtube id
 */
export const getYouTubeId = url => {
  const reShort = /https:\/\/youtu.be\/(.*)/;
  const reLong = /https:\/\/www.youtube.com\/watch\?v=(.*)/;
  const idShort = url.match( reShort );
  const idLong = url.match( reLong );
  if ( idShort ) {
    return idShort[1];
  } if ( idLong ) {
    return idLong[1];
  }
  return null;
};
