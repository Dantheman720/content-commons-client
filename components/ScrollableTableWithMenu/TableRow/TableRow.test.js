import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { formatDate } from 'lib/utils';
import TableRow from './TableRow';

/**
 * Need to mock Next.js dynamic imports
 * in order for this test suite to run.
 */
jest.mock( 'next-server/dynamic', () => () => 'VideoDetailsPopup' );
jest.mock( 'next-server/dynamic', () => () => 'ImageDetailsPopup' );

const props = {
  d: {
    id: 'd137',
    createdAt: formatDate( '2019-05-09T18:33:03.368Z' ),
    updatedAt: formatDate( '2019-05-09T18:33:03.368Z' ),
    team: 'IIP Video Production',
    author: 'Jane Doe',
    projectTitle: 'Test Title 1',
    status: 'PUBLISHED',
    visibility: 'INTERNAL',
    thumbnails: {
      url: 'https://thumbnailurl.com',
      alt: 'some alt text',
    },
    categories: 'about america'
  },
  selectedItems: new Map(),
  tableHeaders: [
    { name: 'projectTitle', label: 'PROJECT TITLE' },
    { name: 'visibility', label: 'VISIBILITY' },
    { name: 'updatedAt', label: 'CREATED' },
    { name: 'team', label: 'TEAM' }
  ],
  toggleItemSelection: jest.fn()
};

const Component = <TableRow { ...props } />;

describe( '<TableRow />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders the persistent table headers', () => {
    const wrapper = shallow( Component );
    const columns = wrapper.find( '.items_table_item' );
    const columnCount = columns.length;
    const expectedColumnCount = props.tableHeaders.length;

    expect( columnCount ).toEqual( expectedColumnCount );
    expect( toJSON( columns ) ).toMatchSnapshot();
    columns.forEach( ( column, i ) => {
      const { label } = props.tableHeaders[i];
      const name = props.d[props.tableHeaders[i].name];
      if ( i === 0 ) {
        expect( column.find( 'MyProjectPrimaryCol' ).exists() )
          .toEqual( true );
        expect( column.find( 'TableMobileDataToggleIcon' ).exists() )
          .toEqual( true );
      } else {
        expect( column.children().contains( label ) ).toEqual( true );
        expect( column.children().contains( name ) ).toEqual( true );
      }
    } );
  } );

  it( 'renders Author, Categories, and Modified Date columns when selected', () => {
    const additionalHeaders = [
      { name: 'author', label: 'AUTHOR' },
      { name: 'categories', label: 'CATEGORIES' },
      { name: 'createdAt', label: 'MODIFIED DATE' }
    ];

    const newProps = {
      ...props,
      tableHeaders: [
        ...props.tableHeaders,
        ...additionalHeaders
      ]
    };

    const wrapper = shallow( <TableRow { ...newProps } /> );
    const columns = wrapper.find( '.items_table_item' );
    const columnCount = columns.length;
    const expectedColumnCount = newProps.tableHeaders.length;

    expect( columnCount ).toEqual( expectedColumnCount );
    expect( toJSON( columns ) ).toMatchSnapshot();
    columns.forEach( ( column, i ) => {
      const { label } = newProps.tableHeaders[i];
      const name = newProps.d[newProps.tableHeaders[i].name];
      if ( i === 0 ) {
        expect( column.find( 'MyProjectPrimaryCol' ).exists() )
          .toEqual( true );
        expect( column.find( 'TableMobileDataToggleIcon' ).exists() )
          .toEqual( true );
      } else {
        expect( column.children().contains( label ) ).toEqual( true );
        expect( column.children().contains( name ) ).toEqual( true );
      }
    } );
  } );

  it( 'renders null if props.d is falsy', () => {
    const newProps = { ...props, d: null };
    const wrapper = shallow( <TableRow { ...newProps } /> );

    expect( wrapper.html() ).toEqual( null );
  } );
} );
