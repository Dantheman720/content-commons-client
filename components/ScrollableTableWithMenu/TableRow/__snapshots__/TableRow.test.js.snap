// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow /> renders Author, Categories, and Date columns when selected 1`] = `
Array [
  <TableCell
    as="td"
    className="items_table_item"
    data-header="PROJECT TITLE"
    key="d137_projectTitle"
  >
    <div
      className="primary_col"
    >
      <MyProjectPrimaryCol
        d={
          Object {
            "author": "Jane Doe",
            "categories": "about america",
            "createdAt": "May 9, 2019",
            "id": "d137",
            "projectTitle": "Test Title 1",
            "status": "PUBLISHED",
            "team": "IIP Video Production",
            "thumbnails": Object {
              "alt": "some alt text",
              "url": "https://thumbnailurl.com",
            },
            "updatedAt": "May 9, 2019",
            "visibility": "INTERNAL",
          }
        }
        header={
          Object {
            "label": "PROJECT TITLE",
            "name": "projectTitle",
          }
        }
        selectedItems={Array []}
        toggleItemSelection={[MockFunction]}
      />
    </div>
    <TableMobileDataToggleIcon
      isOpen={false}
      toggleDisplay={[Function]}
    />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="VISIBILITY"
    key="d137_visibility"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        VISIBILITY
      </div>
      INTERNAL
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="CREATED"
    key="d137_updatedAt"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        CREATED
      </div>
      May 9, 2019
    </span>
    <br />
    <span>
      PUBLISHED
    </span>
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="TEAM"
    key="d137_team"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        TEAM
      </div>
      IIP Video Production
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="AUTHOR"
    key="d137_author"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        AUTHOR
      </div>
      Jane Doe
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="CATEGORIES"
    key="d137_categories"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        CATEGORIES
      </div>
      about america
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="DATE"
    key="d137_createdAt"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        DATE
      </div>
      May 9, 2019
    </span>
    <br />
  </TableCell>,
]
`;

exports[`<TableRow /> renders Author, Categories, and Modified Date columns when selected 1`] = `
Array [
  <TableCell
    as="td"
    className="items_table_item"
    data-header="PROJECT TITLE"
    key="d137_projectTitle"
  >
    <div
      className="primary_col"
    >
      <MyProjectPrimaryCol
        d={
          Object {
            "author": "Jane Doe",
            "categories": "about america",
            "createdAt": "May 9, 2019",
            "id": "d137",
            "projectTitle": "Test Title 1",
            "status": "PUBLISHED",
            "team": "IIP Video Production",
            "thumbnails": Object {
              "alt": "some alt text",
              "url": "https://thumbnailurl.com",
            },
            "updatedAt": "May 9, 2019",
            "visibility": "INTERNAL",
          }
        }
        header={
          Object {
            "label": "PROJECT TITLE",
            "name": "projectTitle",
          }
        }
        selectedItems={Array []}
        toggleItemSelection={[MockFunction]}
      />
    </div>
    <TableMobileDataToggleIcon
      isOpen={false}
      toggleDisplay={[Function]}
    />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="VISIBILITY"
    key="d137_visibility"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        VISIBILITY
      </div>
      INTERNAL
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="CREATED"
    key="d137_updatedAt"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        CREATED
      </div>
      May 9, 2019
    </span>
    <br />
    <span>
      PUBLISHED
    </span>
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="TEAM"
    key="d137_team"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        TEAM
      </div>
      IIP Video Production
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="AUTHOR"
    key="d137_author"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        AUTHOR
      </div>
      Jane Doe
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="CATEGORIES"
    key="d137_categories"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        CATEGORIES
      </div>
      about america
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="MODIFIED DATE"
    key="d137_createdAt"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        MODIFIED DATE
      </div>
      May 9, 2019
    </span>
    <br />
  </TableCell>,
]
`;

exports[`<TableRow /> renders the persistent table headers 1`] = `
Array [
  <TableCell
    as="td"
    className="items_table_item"
    data-header="PROJECT TITLE"
    key="d137_projectTitle"
  >
    <div
      className="primary_col"
    >
      <MyProjectPrimaryCol
        d={
          Object {
            "author": "Jane Doe",
            "categories": "about america",
            "createdAt": "May 9, 2019",
            "id": "d137",
            "projectTitle": "Test Title 1",
            "status": "PUBLISHED",
            "team": "IIP Video Production",
            "thumbnails": Object {
              "alt": "some alt text",
              "url": "https://thumbnailurl.com",
            },
            "updatedAt": "May 9, 2019",
            "visibility": "INTERNAL",
          }
        }
        header={
          Object {
            "label": "PROJECT TITLE",
            "name": "projectTitle",
          }
        }
        selectedItems={Array []}
        toggleItemSelection={[MockFunction]}
      />
    </div>
    <TableMobileDataToggleIcon
      isOpen={false}
      toggleDisplay={[Function]}
    />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="VISIBILITY"
    key="d137_visibility"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        VISIBILITY
      </div>
      INTERNAL
    </span>
    <br />
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="CREATED"
    key="d137_updatedAt"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        CREATED
      </div>
      May 9, 2019
    </span>
    <br />
    <span>
      PUBLISHED
    </span>
  </TableCell>,
  <TableCell
    as="td"
    className="items_table_item"
    data-header="TEAM"
    key="d137_team"
  >
    <span>
      <div
        className="items_table_mobileHeader"
      >
        TEAM
      </div>
      IIP Video Production
    </span>
    <br />
  </TableCell>,
]
`;
